File: cpp/Biscuits.cpp
#include "Biscuits.h"

Biscuits::Biscuits(std::string name, double price, int quantity, std::chrono::system_clock::time_point expiryDate, double weight)
    : ExpirableProduct(name, price, quantity, expiryDate), weight(weight) {}

double Biscuits::getWeight() const {
    return weight;
}

bool Biscuits::requiresShipping() const {
    return true;
}

File: javascript/Biscuits.js
// javascript/Biscuits.js
import { ExpirableProduct } from './ExpirableProduct.js';
import { IShippable } from './Shippable.js';

export class Biscuits extends ExpirableProduct {
    constructor(name, price, quantity, expiryDate, weight) {
        super(name, price, quantity, expiryDate);
        this.weight = weight;
    }

    getWeight() {
        return this.weight;
    }

    requiresShipping() {
        return true;
    }
}
